lib/papertrail/cli.rb

```diff
@@ -137,12 +142,31 @@ def query_time_range
       end
     end
 
+    COLORS = [:cyan, :yellow, :green, :magenta, :red]
+
+    def colorize(event)
+      idx = (event.data["hostname"].length + event.data["program"].length) % 5
```

Comment:
This is a fantastic comment, and I was also wondering whether I'd like it. I encountered a similar problem where related hosts' output was the same color (your 1st example), but I actually preferred that output over many different colors (your 2nd example).

That said, I can imagine lots of cases where (a) the hostname+program lengths would be unintentionally overlapping, or (b) the overlap did indicate a relationship like `webN`, but it wasn't the one I wanted.

What about combining these with another gap you identified, and changing `--force-colors` into `--colors={h,p,hp,none}`? The default would be `--colors=hp` (your 2nd example) when color terminal emulation was detected, but someone could also use `--colors=p` for the same behavior as Heroku's CLI (and your 1st example) or `--colors=h` to make it easier to spot multiline output from the same sender.

One important thing to note is that we only have 5 characters, so color collisions are going to happen anyway. We can try to make them less of an issue, though.